### PR TITLE
Support dynamic chunk size for uploads

### PR DESCRIPTION
**Pull request by jaredsmith**
Add A/B test middleware, and start interfacing with UpChunk

### CODE REVIEW

**Comment by jaredsmith** [comment]
This should now be working for browser-recorded video.  I'll tackle uploaded video (with `mux-uploader/mux-uploader-react`) next.

**Review by decepulis — APPROVED** [review]

**Comment by jaredsmith** [comment]
This should also now work for `<mux-uploader-react/>` (which calls `<mux-uploader/>`) now too, as soon as https://github.com/muxinc/elements/pull/337 is merged and new builds are created there.

**Comment by jaredsmith** [comment]
As soon as #138 is merged, I'll make sure this applies cleanly (and CI tests pass cleanly) and is ready to merge as well.

File: pages/_middleware.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import type { NextRequest } from 'next/server';
+
+const COOKIE_KEY = 'dynamicChunkSize';
+
+export function middleware(request: NextRequest): NextResponse {
+  const response = NextResponse.next();
+  const cookieFromRequest = request.cookies[COOKIE_KEY];
+
+  if (!cookieFromRequest) {
+    const selectedForExperiment = Math.random() <= 0.2;
```
Comment (jaredsmith) [inline pages/_middleware.tsx:11]: A/B testing is currently set to 20% -- we can change this over time once we're more confident in the dynamic chunk sizes.

**Review by jsanford8 — APPROVED** [review]

**Pull request merged**